### PR TITLE
Dynamic configuration and live debugger

### PR DESCRIPTION
**Pull request by bwoebi**

### CODE REVIEW

File: ext/live_debugger.c [hunk 1]
```diff
@@ -0,0 +1,1286 @@
...
+
+    dyn->payload = NULL;
+    dyn->rejected = def->parent.probe.evaluate_at == DDOG_EVALUATE_AT_ENTRY && !dd_log_probe_eval_condition(def, execute_data, &retval);
+    dyn->capture_arena = NULL;
+
+    if (!dyn->rejected && def->parent.probe.evaluate_at == DDOG_EVALUATE_AT_ENTRY) {
+        dd_log_probe_ensure_payload(dyn, def, NULL);
+        if (def->parent.probe.probe.log.capture_snapshot) {
+            ddog_DebuggerCapture *capture = ddog_snapshot_entry(dyn->payload);
+            DDTRACE_G(debugger_capture_arena) = zend_arena_create(65536);
```
Comment (iamluc) [inline ext/live_debugger.c:473]: `DDTRACE_G(debugger_capture_arena)` can also be created in `ddtrace_collect_exception_debug_data`. Isn't there a risk of replacing the arena created there?

File: ext/sidecar.c [hunk 2]
```diff
@@ -77,20 +76,18 @@ ddog_SidecarTransport *dd_sidecar_connection_factory(void) {
...
-                                    // Not used yet
-                                    NULL,
-                                    remote_config_products,
-                                    0,
-                                    remote_config_capabilities,
-                                    0
-                                    );
+                                    ddtrace_set_all_thread_vm_interrupt,
+                                    DDTRACE_REMOTE_CONFIG_PRODUCTS,
+                                    sizeof(DDTRACE_REMOTE_CONFIG_PRODUCTS) / sizeof(DDTRACE_REMOTE_CONFIG_PRODUCTS[0]) - !get_DD_DYNAMIC_INSTRUMENTATION_ENABLED(),
```
Comment (iamluc) [inline ext/sidecar.c:88]: It's probably worth adding a comment where `DDTRACE_REMOTE_CONFIG_PRODUCTS` is declared, indicating that `RemoteConfigProduct::LiveDebugger` must remain the last element.

**Review by iamluc — APPROVED** [review]
:clap:  Really impressive PR :astonished:

File: components-rs/remote_config.rs [hunk 3]
```diff
@@ -0,0 +1,409 @@
...
+        }
+    } else {
+        true
+    }
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn ddog_CharSlice_to_owned(str: CharSlice) -> *mut Vec<c_char> {
+    Box::into_raw(Box::new(str.as_slice().into()))
+}
```
[the remark this answers is not in the document]
Comment (bwoebi) [reply components-rs/remote_config.rs:330]: Not sure, this returns a proper rust vec... I'd move that when needed anywhere else.

File: ext/live_debugger.c (hunk 1 above)
Comment (bwoebi) [reply ext/live_debugger.c:473]: This function is created called during serialization and dropped after serialization. No function calls should happen here.

File: ext/sidecar.c (hunk 2 above)
Comment (bwoebi) [reply ext/sidecar.c:88]: Agree.

**Pull request merged**